# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/DeviceManagerDx.cpp
# Commit: 87f3532bfbcd
# Full Hash: 87f3532bfbcd50706f16c61eff4c6bdd93fe34aa
# Author: Kelsey Gilbert <kelsey.gilbert@mozilla.com>
# Date: 2023-02-15 18:16:35
# Regressor Bug: 1799258
# File Overlap Count: 1
# Description:
#   Bug 1799258 - Share all-of-dcomp.h preamble, and deal with outdated mingw dcomp.h. r=gfx-reviewers,sotaro
#   
#   Mingw's dcomp.h is not the official one, but rather a by-hand
#   reproduction. While this newly-updated version has e.g.
#   IDCompositionFilterEffect, it is still missing e.g.
# ==============================================================================

diff -r 9c1d9405e8bf -r 87f3532bfbcd gfx/thebes/DeviceManagerDx.cpp
--- a/gfx/thebes/DeviceManagerDx.cpp	Wed Feb 15 08:29:43 2023 +0000
+++ b/gfx/thebes/DeviceManagerDx.cpp	Wed Feb 15 08:29:43 2023 +0000
@@ -25,17 +25,10 @@
 #include "nsPrintfCString.h"
 #include "nsString.h"
 
-#undef _WIN32_WINNT
-#define _WIN32_WINNT _WIN32_WINNT_WINBLUE
-#undef NTDDI_VERSION
-#define NTDDI_VERSION NTDDI_WINBLUE
+// -
 
-// We also need this, or dcomp.h won't give us e.g. IDCompositionDevice3:
-#undef _WIN32_WINNT_WINTHRESHOLD
-#define _WIN32_WINNT_WINTHRESHOLD _WIN32_WINNT
-
+#include "mozilla/gfx/AllOfDcomp.h"
 #include <d3d11.h>
-#include <dcomp.h>
 #include <ddraw.h>
 #include <dxgi.h>
 